soco/core.py

```diff
@@ -1490,6 +1493,59 @@ def get_item_album_art_uri(self, item):
         else:
             return None
 
+    @only_on_master
+    def set_sleep_timer(self, sleep_time_seconds):
+        """Sets the sleep timer.
+
+        Args:
+            sleep_time_seconds (`int`): How long to wait before turning off
+                speaker in seconds. Maximum value of 86399
+
+        Returns:
+            bool: True if succesful, False otherwise
+
+        Raises SoCoException (or a subclass) upon errors, ValueError for
+            incorrect syntax.
+
+        """
+        if sleep_time_seconds and \
```

Comment:
Wait! As I read your comment two steps down, I realize, now I know how you're thinking this can be restructured . . . lemme know what you think.